jinahub/rankers/CatboostRanker/catboost_ranker.py

```diff
@@ -0,0 +1,167 @@
+        """
+        data, label, group_id = self._extract_features(docs)
+        if self.weight:
+            return Pool(
+                data=data,
+                label=label,
+                group_id=group_id,
+                weight=self._extract_weights(docs),
+            )
+        else:
+            return Pool(data=data, label=label, group_id=group_id)
+
+    @requests(on='/train')
+    def train(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """This endpoint fits a catboost ranker based on docs.
+
+        :param docs: Documents to fit the model, each docs should have fields corresponded to
+          the :attr:`query_features` and has matches with corresponded :attr:`match_features` to
+          extract feature from. Also, a label field should be stored in each match document.
+        :param parameters: Parameters to set, we'll get :attr:`catboost_parameters` from the parameters field.
+        """
+        catboost_parameters = parameters.get(
+            'catboost_parameters', self.catboost_parameters
+        )
+        train_pool = self.build_catboost_pool(docs)
+        self.model = CatBoostRanker(**catboost_parameters)
+        self.model.fit(train_pool)
+
+    @requests(on='/search')
+    def rank(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """This endpoint ranks a catboost ranker based on docs.
+
+        :param docs: Documents to fit the model, each docs should have fields corresponded to
+          the :attr:`query_features` and has matches with corresponded :attr:`match_features` to
+          extract feature from. An additional field will be attached to the matched documents as was defined
+          as :attr:`label`.
+        :param parameters: Parameters to set, we'll get :attr:`catboost_parameters` from the parameters field.
+        """
+        feature_vecs, _, _ = self._extract_features(docs)
+        if not self.model.is_fitted():
+            raise ValueError(
+                'You need to train your model before make prediction, call `train` endpoint first.'
+            )
+        preds = self.model.predict(feature_vecs)
+        matches = docs.traverse_flat(traversal_paths=['m'])
+        for match, pred in zip(matches, preds):
+            match.scores[self.label] = pred
+        for doc in docs:
+            doc.matches.sort(key=lambda x: x.scores[self.label].value, reverse=True)
+
+    @requests(on='/dump')
+    def dump(self, parameters: Dict, **kwargs):
+        """Dump trained model to specified path
+
+        :param parameters: Parameters pass to this endpoint, expect :attr:`model_path` to be set.
+        """
+        model_path = parameters.get('model_path', None)
+        if model_path:
+            self._save_model(model_path)
+        else:
+            raise ValueError('Please specify a `model_path` inside parameters variable')
+
+    @requests(on='/load')
+    def load(self, parameters: Dict, **kwargs):
```

Comment:
Do we need the `load` endpoint at all? I don't see why anyone would want to load a model after it has already been loaded for the first time

Reply:
i'd like to add it, user might need to deploy multiple flows, dump here and load there etc.

Reply:
I see